server/src/main/java/com/broadcom/lsp/cobol/core/strategy/CobolErrorStrategy.java

```diff
@@ -36,18 +42,41 @@
   private static final String REPORT_NO_VIABLE_ALTERNATIVE = "reportNoViableAlternative";
   private static final String REPORT_UNWANTED_TOKEN = "reportUnwantedToken";
   private static final String REPORT_MISSING_TOKEN = "reportMissingToken";
+  private static final String STARTING_CURLY_BRACKET = "{";
+  private static final String ENDING_CURLY_BRACKET = "}";
+  private static final String ERROR_DELIMITER = ", ";
+  private static final String CURLY_BRACKET_REGEX = "[{}]";
+  private static Map<String, String> charTokensCacheMap = new ConcurrentHashMap<>();
 
   private MessageService messageService;
+  private CompletionStorage keywords;
 
   @Inject
-  public CobolErrorStrategy(MessageService messageService) {
+  public CobolErrorStrategy(
+      MessageService messageService, @Named("Keywords") CompletionStorage keywords) {
     this.messageService = messageService;
+    this.keywords = keywords;
   }
 
   private static String getRule(Parser recognizer) {
     return recognizer.getRuleInvocationStack().get(0);
   }
 
+  private static void setTokensMapCache(Vocabulary vocabulary) {
+    charTokensCacheMap =
+        IntStream.range(1, vocabulary.getMaxTokenType())
+            .boxed()
+            .filter(element -> Objects.nonNull(vocabulary.getLiteralName(element)))
+            .collect(Collectors.toMap(vocabulary::getSymbolicName, vocabulary::getLiteralName));
+    // Special case for DOT_FS. Its not a tokens for characters.
+    charTokensCacheMap.put("DOT_FS", "'.'");
```

Comment:
DOT_FS : '.' ('\r' | '\n' | '\f' | '\t' | ' ')+ | '.' EOF;
It's not considered tokens for characters but is a rule.

Reply:
I see that there are a lot of similar declarations, e.g. COMMA_EOF, DOLLARCHAR, LEVEl_NUMBER, etc.
I don't see an automatic way how to get the actual value by name, maybe it requires defining a map with those technical token rule names and the actual value of them. So, here the meaningful part of the value is '.', and for level number, it is '01-49, 66, 77, 88'